src/charm.py

```diff
@@ -135,15 +154,26 @@ def _configure(self, event: EventBase) -> None:
             usim_sequence_number=self._get_usim_sequence_number_from_config(),  # type: ignore[arg-type]  # noqa: E501
             sst=self._get_sst_from_config(),  # type: ignore[arg-type]
             tac=self._get_tac_from_config(),  # type: ignore[arg-type]
-            upf_gateway=self._get_upf_gateway_from_config(),  # type: ignore[arg-type]
-            upf_ip_address=self._get_upf_ip_address_from_config(),  # type: ignore[arg-type]
+            upf_gateway=upf_gateway,
+            upf_ip_address=upf_ip_address,
```

Comment:
To check: if the upf gatway and ip address don't exist these parameters will be empty in config file.